Release candidate changes

gonzaw opened the pull request:
Final touches before v1.0 release:
* README updated with new examples and description / usage
* sample references DLLs directly and not the nuget package

Commit:
Make final updates before 1.0 release

- Makes the sample reference the DLLs directly (referencing the VS projects inside its solution) instead of referencing the NuGet package
- Updates the NuGet version to v1.0.0

Commit:
Update README.MD

-Updates the readme, adding more examples, ranging from simple to complex ones

rrivem commented on README.md:
```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+For a more complex example, imagine "GET http://api.catsforall.com/cats/myCatName" returns a _404 Not Found_ status code if the cat doesn't exist. With our current API, in that scenario we throw an exception. However, what if we want to handle that case programatically? We can do that in the following way:
+
+![restclient example 1](docs/img/restclient_cat_example_get_option.png?raw=true)
```
let's put first the OptionAsNullProcessor, so can explain the usage of processors first, before digging into Monad

rrivem commented on README.md:
```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+The only thing that changed from the original method is the return type, which changes from `Cat` to `OptionStrict<Cat>`, and we added a line `AddProcessors(new OptionAsNotFoundProcessor<Cat>())`. 
+
+Let's talk about the change in the returning type first. `OptionStrict<T>` is a type from the [Csharp-Monad](https://github.com/louthy/csharp-monad) library, it represents a nullable type (which can be either a value type or reference type). A value with type `OptionStrict<T>` is either empty or has a value. We can use this type to represent our new method: If the API call returns a 200 status co ...
```
which I probably resume (advantages over null comparison) before showing the code that uses the OptionAsNotFoundProcessor

rrivem commented on README.md:
```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+## Managing 404 status codes
+
+For a more complex example, imagine "GET http://api.catsforall.com/cats/myCatName" returns a _404 Not Found_ status code if the cat doesn't exist. With our current API, in that scenario we throw an exception. However, what if we want to handle that case programatically? We can do that in the following way:
```
Don't imagine, the api behaves like that, save words... (and cats)

rrivem commented on README.md:
```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+## Processors
+
+rest-client uses a pipeline of processors that check whether they find a specific HTTP status code, and if they find it they return a value of the returning type.
```
here we need an image to illustrate the processors pipeline

rrivem commented on README.md:
```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+## Managing 4xx / 5xx status codes
+
+With rest-client we can handle an even more complex subject, like managing HTTP errors with C# types. Until know, every 4xx and 5xx error was not known to our program, i.e it was not represented in the return type. Instead, if the API returned such status codes, our program thre and exception and then forgot about it. There exist cases where we want our own program to handle such errors programat ...
```
Monad is just a flavor of our like, you can use this library without using Monad. Every example we present with Monad should show an advantage of usage try-catch vs if or Either.Select

Commit:
Merge branch 'master' into release_candidate_changes

# Conflicts:
#	README.md

Commit:
Updated README and code

- Updated README with code syntax instead of images
- Renaming of POST/PUT/etc default operations

Commit:
Update README and default implementations of PUT/POST/DELETE
